src/common/stores/emote-menu-view-store.js

```diff
@@ -112,6 +116,26 @@ class EmoteMenuViewStore extends SafeEventEmitter {
     this.markDirty(false);
   }
 
+  registerProvider(provider) {
+    const registeredProviderId = uuidv4();
+    registeredProviders[registeredProviderId] = provider;
+    return registeredProviderId;
+  }
+
+  getRegisteredProvider(providerId) {
+    return registeredProviders[providerId];
+  }
+
+  upsertRegisteredProviderCategory(categoryId, category) {
```

Comment:
should `id` be part of the category?

Reply:
this hasn't been addressed, so marking unresolved.

`upsertRegisteredProviderCategory(category)` should be the function declaration